Sources/BrowserServicesKit/SecureVault/SecureVaultManager.swift

```diff
@@ -158,5 +150,64 @@ extension SecureVaultManager: AutofillSecureVaultDelegate {
             completionHandler(nil)
         }
     }
+    
+    func existingEntries(for domain: String, autofillData: AutofillUserScript.DetectedAutofillData) throws -> AutofillData {
+        
+        let vault = try SecureVaultFactory.default.makeVault(errorReporter: self.delegate)
+        
+        // Determine if the identity should be sent to the client app:
+
+        var proposedIdentity: SecureVaultModels.Identity?
+        
+        if let identity = autofillData.identity, try vault.existingIdentityForAutofill(matching: identity) == nil {
+            os_log("Got new identity/address to save", log: .passwordManager, type: .info)
+            proposedIdentity = identity
+        } else {
+            os_log("No new identity/address found, avoid prompting user", log: .passwordManager, type: .info)
+        }
+        
+        // Determine if the credentials should be sent to the client app:
+        
+        var proposedCredentials: SecureVaultModels.WebsiteCredentials?
+
+        if let credentials = autofillData.credentials, let passwordData = credentials.password.data(using: .utf8) {
+            if let account = try vault
+                .accountsFor(domain: domain)
+                .first(where: { $0.username == credentials.username }) {
+                
+                if let existingAccountID = account.id,
+                   let existingCredentials = try vault.websiteCredentialsFor(accountId: existingAccountID),
+                   existingCredentials.password == passwordData {
+                    os_log("Found duplicate credentials, avoid prompting user", log: .passwordManager, type: .info)
+                    proposedCredentials = nil
+                } else {
+                    os_log("Found existing credentials to update", log: .passwordManager, type: .info)
+                    proposedCredentials = SecureVaultModels.WebsiteCredentials(account: account, password: passwordData)
+                }
+
+            } else {
+                os_log("Received new credentials to save", log: .passwordManager, type: .info)
+                let account = SecureVaultModels.WebsiteAccount(username: credentials.username ?? "", domain: domain)
+                proposedCredentials = SecureVaultModels.WebsiteCredentials(account: account, password: passwordData)
+            }
+        } else {
+            os_log("No new credentials found, avoid prompting user", log: .passwordManager, type: .info)
+        }
```

Comment:
This has gotten a bit out of hand, I'll try to clean it up. The Autofill functionality has changed in such a way that it doesn't try to protect against storeData calls directly after you've autofilled, which means BSK now has to do duplicate checks around credentials.